### PR TITLE
Render grouped activities correctly and redraw without browser refresh when a new file is uploaded.

### LINKED ISSUE #3510: Timeline plan view should display new activities immediately
When a new JSON file is uploaded into an existing Plan view, the view should re-render and display the new plan without needing a browser refresh.
Currently a browser refresh is required.

### LINKED ISSUE #3511: Timeline Plan view does not group activities correctly
Activities from groups seem to bleed into each other. Each group's activities should be rendered within that group.

### PR DESCRIPTION
**Pull request by shefalijoshi**
Resolves #3511 Ensure that overlap checking only looks at activities within it's own group. This is done by assuming that any rows less than a group's starting row belong to another group.

Resolves #3510 Observe for changes to a plan and update the plan view accordingly.

## Author Checklist
Changes address original issue? Yes
Unit tests included and/or updated with changes? N - Not sure there's a way to test this as it is drawing logic.
Command line build passes? Yes
Changes have been smoke-tested? Yes
Testing instructions included? Yes

### CODE REVIEW

File: src/plugins/timeline/Plan.vue [hunk 1]
```diff
@@ -61,11 +61,18 @@ export default {
         this.openmct.time.on("timeSystem", this.setScaleAndPlotActivities);
         this.openmct.time.on("bounds", this.updateViewBounds);
         this.resizeTimer = setInterval(this.resize, RESIZE_POLL_INTERVAL);
+        this.unlistenFileChange = this.openmct.objects.observe(this.domainObject, '*', (mutatedObject) => {
+            this.validateJSON(mutatedObject.selectFile.body);
+            this.setScaleAndPlotActivities();
+        });
```
Comment (deeptailor) [inline src/plugins/timeline/Plan.vue:67]: can we rename this, and make the callback a method? It would be easier to read

**Review by deeptailor — CHANGES_REQUESTED** [review]
Just one comment, otherwise looks good

File: src/plugins/timeline/Plan.vue (hunk 1 above)
Comment (shefalijoshi) [reply src/plugins/timeline/Plan.vue]: Done.

**Review by deeptailor — APPROVED** [review]
LGTM

**Comment by deeptailor** [comment]
## Reviewer Checklist
Changes appear to address issue? Yes
Appropriate unit tests included? N/A
Code style and in-line documentation are appropriate? Yes
Commit messages meet standards? Yes
Has associated issue been labelled unverified? (only applicable if this PR closes the issue) Yes

**Pull request merged**